src/pages/workspace/categories/CreateCategoryPage.tsx

```diff
@@ -0,0 +1,107 @@
+import {Keyboard} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapper from '@components/Form/InputWrapper';
+import type {FormInputErrors, FormOnyxValues} from '@components/Form/types';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import TextInput from '@components/TextInput';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import type {CentralPaneNavigatorParamList} from '@navigation/types';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import PaidPolicyAccessOrNotFoundWrapper from '@pages/workspace/PaidPolicyAccessOrNotFoundWrapper';
+import * as Policy from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import INPUT_IDS from '@src/types/form/WorkspaceCategoryCreateForm';
+import type {PolicyCategories} from '@src/types/onyx';
+
+type WorkspaceCreateCategoryPageOnyxProps = {
+    /** All policy categories */
+    policyCategories: OnyxEntry<PolicyCategories>;
+};
+
+type CreateCategoryPageProps = WorkspaceCreateCategoryPageOnyxProps & StackScreenProps<CentralPaneNavigatorParamList, typeof SCREENS.WORKSPACE.CATEGORY_CREATE>;
+
+function CreateCategoryPage({route, policyCategories}: CreateCategoryPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const validate = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.WORKSPACE_CATEGORY_CREATE_FORM>) => {
+            const errors: FormInputErrors<typeof ONYXKEYS.FORMS.WORKSPACE_CATEGORY_CREATE_FORM> = {};
+            const categoryName = values.categoryName.trim();
+
+            if (!ValidationUtils.isRequiredFulfilled(categoryName)) {
+                errors.categoryName = 'workspace.categories.categoryRequiredError';
+            } else if (policyCategories?.[categoryName]) {
+                errors.categoryName = 'workspace.categories.existingCategoryError';
+            } else if (categoryName === CONST.INVALID_CATEGORY_NAME) {
+                errors.categoryName = 'workspace.categories.invalidCategoryName';
+            } else if ([...categoryName].length > CONST.CATEGORY_NAME_LIMIT) {
+                // Uses the spread syntax to count the number of Unicode code points instead of the number of UTF-16
+                // code units.
+                ErrorUtils.addErrorMessage(errors, 'categoryName', ['common.error.characterLimitExceedCounter', {length: [...categoryName].length, limit: CONST.TITLE_CHARACTER_LIMIT}]);
```

Comment:
```suggestion
                ErrorUtils.addErrorMessage(errors, 'categoryName', ['common.error.characterLimitExceedCounter', {length: [...categoryName].length, limit: CONST.CATEGORY_NAME_LIMIT}]);
```

We're using `category_name_limit` in the condition and `title_character_limit` in the error message.

Reply:
Updated